src/blame.rs

```diff
@@ -0,0 +1,289 @@
+        unsafe { (*self.raw).final_start_line_number as usize }
+    }
+
+    /// Returns the OID of the commit where this hunk was found.
+    ///
+    /// This will usually be the same as `final_commit_id`,
+    /// except when `GIT_BLAME_TRACK_COPIES_ANY_COMMIT_COPIES`
+    /// has been specified
+    pub fn orig_commit_id(&self) -> Oid {
+        unsafe { Oid::from_raw(&(*self.raw).orig_commit_id) }
+    }
+
+    /// Returns signature of the commit.
+    pub fn orig_signature(&self) -> Signature {
+        unsafe { signature::from_raw_const(self, (*self.raw).orig_signature) }
+    }
+
+    /// Returns line number where this hunk begins.
+    ///
+    /// Note that the start line is counting from 1.
+    pub fn orig_start_line(&self) -> usize {
+        unsafe { (*self.raw).orig_start_line_number as usize }
+    }
+
+    /// Returns path to the file where this hunk originated.
+    pub fn path(&self) -> &Path {
+        unsafe {
+            util::bytes2path(::opt_bytes(self, (*self.raw).orig_path).unwrap())
+        }
+    }
+
+    /// Tests whether this hunk has been tracked to a boundary commit.
+    pub fn is_boundary(&self) -> bool {
+        unsafe {
+            (*self.raw).boundary == 1
+        }
+    }
+
+    /// Returns number of lines in this hunk.
+    pub fn len(&self) -> usize {
+        unsafe {
+            (*self.raw).lines_in_hunk as usize
+        }
+    }
+}
+
+impl BlameOptions {
+
+    /// Initialize options
+    pub fn new() -> BlameOptions {
+        let mut opts = BlameOptions {
+            raw: unsafe { mem::zeroed() },
+        };
+        assert_eq!(unsafe {
+            raw::git_blame_init_options(&mut opts.raw,
+                                        raw::GIT_BLAME_OPTIONS_VERSION)
+        }, 0);
+        opts
+    }
+
+    fn flag(&mut self, opt: u32, val: bool) -> &mut BlameOptions {
+        if val {
+            self.raw.flags |= opt;
+        } else {
+            self.raw.flags &= !opt;
+        }
+        self
+    }
+
+    /// Same file
+    pub fn track_copies_same_file(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_TRACK_COPIES_SAME_FILE, opt)
+    }
+
+    /// Same commit moves
+    pub fn track_copies_same_commit_moves(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_TRACK_COPIES_SAME_COMMIT_MOVES, opt)
+    }
+
+    /// Same commit copies
+    pub fn track_copies_same_commit_copies(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_TRACK_COPIES_SAME_COMMIT_COPIES, opt)
+    }
+
+    /// Any commit copies
+    pub fn track_copies_any_commit_copies(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_TRACK_COPIES_ANY_COMMIT_COPIES, opt)
+    }
+
+    /// First parent
+    pub fn first_parent(&mut self, opt: bool) -> &mut BlameOptions {
+        self.flag(raw::GIT_BLAME_FIRST_PARENT, opt)
+    }
+
+    /// Raw pointer
+    pub unsafe fn raw(&mut self) -> *mut raw::git_blame_options {
```

Comment:
Gah that probably shouldn't be a public function! I'd like to perhaps expose the `Binding` trait one day to lower the representation of each type to the underlying libgit2 type, but I don't think that's quite necessary just yet.